pyproject.toml

```diff
@@ -79,6 +79,8 @@ ml = [
     "safetensors>=0.3.0",
     "transformers>=4.0.0,<5.0.0",
     "accelerate>=0.20.3,<1.0.0",
+    "optuna>=4.0.0",
+    "plotly>=5.18.0", # required by optuna viz
```

Comment:
Do you think we should move this to the `dev` group? some users may not want to install optuna and prefer to train a model ? optuna has several dependencies that we might let the user decide if they want to install them or not.